volumes.go

```diff
@@ -33,8 +33,10 @@ type Volume struct {
 
 // SnapshotPolicy used to execute actions on volume
 type SnapshotPolicy struct {
-	ID   string `json:"id"`
-	Href string `json:"href"`
+	ID                string `json:"id"`
+	Href              string `json:"href"`
+	SnapshotFrequency string `json:"snapshot_frequency,omitempty"`
+	snapshot_count    int    `json:"snapshot_count,omitempty"`
```

Comment:
Without a leading capital letter, the `json` package won't do anything with this field. You should either rename it `SnapshotCount` or get rid of it.